hcl/hcl_test.go

```diff
@@ -132,30 +134,38 @@ module "bleh" {
 			},
 		},
 		{
-			name: "ignore if source is not a string",
+			name: "fails if source is not a string",
 			input: `
 module "test" {
 	source = -1
 }
 `,
 			want: want{
-				err: hcl.ErrMalformedTerraform,
+				err: errors.E(hcl.ErrTerraformSchema, mkrange(start(3, 10, 27), end(3, 13, 29))),
 			},
 		},
 		{
 			name:  "variable interpolation in the source string - fails",
 			input: "module \"test\" {\nsource = \"${var.test}\"\n}\n",
 			want: want{
-				err: hcl.ErrMalformedTerraform,
+				err: errors.E(hcl.ErrTerraformSchema, mkrange(start(2, 9, 28), end(2, 23, 31))),
 			},
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			path := test.WriteFile(t, "", "main.tf", tc.input)
+			configdir := t.TempDir()
+			tfpath := test.WriteFile(t, configdir, "main.tf", tc.input)
 
-			modules, err := hcl.ParseModules(path)
-			assert.IsError(t, err, tc.want.err)
+			if tc.want.err != nil {
+				if e, ok := tc.want.err.(*errors.Error); ok {
+					if e.FileRange.Filename != "" {
+						e.FileRange.Filename = tfpath
+					}
```

Comment:
hmm if the wanted error defines a filename in the range we overwrite it, if none is defined (empty string) we leave it empty. Is this the behavior we want here ? (usually we set automatically something when none is defined, like a default)

Reply:
fixed